### DISCUSSION TITLE
How do i use this wonderful tool ?

### DISCUSSION THREAD

**Discussion by ghost** [jtdor/build-deb-action#5] [General]
Can anyone please tell me how to use this beautiful tool ??, I need to build deb packages for Ubuntu from a .tar.gz source file, using dependencies from ubuntu 22.04 (Latest LTS), how do i do it, i tried uploading packages but the action always fail, please guide me..

**Comment by jtdor** [comment]
Hi there, I’ve converted your question into a discussion, since it’s not really an issue with the action.

Helping you with packaging a new software might be a bit out of scope of a discussion here. You probably get more and faster help asking in a Debian- or Ubuntu-related forum or mailing list.

To use this action, you must have “debianized” the source of the tool you want to package. If you have done that, using the action behaves mostly the same as calling the packaging tools like `dpkg-buildpackage`, `debuild`, `sbuild`, or `pbuilder` manually in your terminal (you will have read about them when you are ready to use the action).

Debianizing a package means you provide a subdirectory `debian/` in your software’s sources that contains a set of standard rules and information about how to build and package the software for Debian or Ubuntu.

Some introductory links about packaging for Debian or Ubuntu:
* [Guide for Debian Maintainers](https://www.debian.org/doc/manuals/debmake-doc/) (see the [Simple Example](https://www.debian.org/doc/manuals/debmake-doc/ch04))
* [Debian New Maintainers' Guide](https://www.debian.org/doc/manuals/maint-guide/)
* [Debian Packaging Tutorial](https://www.debian.org/doc/manuals/packaging-tutorial/packaging-tutorial)
* [Packaging/Intro - Debian Wiki](https://wiki.debian.org/Packaging/Intro)

The general process is the same for both, since Ubuntu is based on Debian.

Feel free to ask for other pointers but no guarantee that I can answer here quickly ;-)
